luaui/Include/blueprint_substitution/reports.lua

```diff
@@ -0,0 +1,182 @@
+-- luaui/Include/blueprint_substitution/reports.lua
+-- Contains reporting functions for api_blueprint.lua
+
+local M = {}
+
+-- Store references passed in during include
+local substitutionLogic = nil
+
+-- Local data for reports
+local sideTotals = {}
+local sideComplete = {}
+local uncategorizedUnits = {}
+
+local function escapeCsvField(value)
```

Comment:
Could eventually go into some sort of `Spring.Utilities`, though not in the scope of this PR.